agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +169,29 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) installerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+	installScript, uninstallScript, err := r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret.Name, byoHost.Spec.InstallationSecret.Namespace)
```

Comment:
Suggestion: `getInstallationScript` would need only the installation secret to be present. So the func signature can be changed to `r.getInstallationScript(ctx, byoHost.Spec.InstallationSecret)`